crm_lead_vat/models.py

```diff
@@ -22,9 +22,10 @@ def _lead_create_contact(self, lead, name, is_company, parent_id=False):
     def on_change_partner_id(self, partner_id):
```

Comment:
Although the fix is good, I feel the real fix to your error would be adding `@api.multi` for this method...